tests/aggregation_regress/tests.py

```diff
@@ -1827,6 +1827,27 @@ def test_having_subquery_select(self):
         )
         self.assertEqual(set(books), {self.b1, self.b4})
 
+    def test_aggregate_and_annotate_duplicate_columns(self):
+        books = (
+            Book.objects.values("isbn")
+            .annotate(
+                name=F("publisher__name"),
+                num_authors=Count("authors"),
+            )
```

Comment:
The test I had in the ticket had two separate annotations to be grouped on; one with `name` as its alias, and one with an original field called `name`. This query has only one field serving both roles, and that does not generate a conflict.

I think this should be made more like I wrote in the ticket:
```suggestion
            Book.objects.values("isbn")
            .annotate(
                name=F("publisher__name"),
                contact_name=F("contact__name"),
                num_authors=Count("authors"),
            )
```

Reply:
Adding two annotations is not required to reproduce the issue; the ambiguity is also present if `publisher.name` is selected as it collides with the `name` alias.

We usually add tests represent the minimal set of criteria to reproduce the issue hence why I dropped the `contact_name=F("contact__name")` annotation from the test.

Reply:
I just ran the tests in the original project where this came from with this branch, and they failed in the same way as they did originally. Checking further.